src/lib/crypto/ecdh.cpp

```diff
@@ -261,9 +276,20 @@ ecdh_encrypt_pkcs5(rng_t *                  rng,
         goto end;
     }
 
-    out->p.len = sizeof(out->p.mpi);
-    if (botan_pk_op_key_agreement_export_public(eph_prv_key, out->p.mpi, &out->p.len)) {
-        goto end;
+    /* we need to prepend 0x40 for the x25519 */
+    if (key->curve == PGP_CURVE_25519) {
+        out->p.len = sizeof(out->p.mpi) - 1;
+        if (botan_pk_op_key_agreement_export_public(
+              eph_prv_key, out->p.mpi + 1, &out->p.len)) {
```

Comment:
For x25519 does there need to be a byte reversal here?

Reply:
No, it doesn't seem so, at least it is interoperable with GnuPG. Looks like 0x40 is considered to tell 'treat data as it is'. Will write to mailing list to ensure this since there is no any information in the 4880-bis.

Reply:
...ok, it was already discussed in the following thread: https://lists.gnupg.org/pipermail/gnupg-devel/2018-February/033437.html

Reply:
@randombit Was this point satisfied?